src/core/qgsogcutils.cpp

```diff
@@ -1931,15 +1931,31 @@ QDomElement QgsOgcUtils::expressionNodeToOgcFilter( const QgsExpression::Node* n
 
 QDomElement QgsOgcUtils::expressionUnaryOperatorToOgcFilter( const QgsExpression::NodeUnaryOperator* node, QDomDocument& doc, QString& errorMessage )
 {
+
+  QDomElement operandElem = expressionNodeToOgcFilter( node->operand(), doc, errorMessage );
+  if ( !errorMessage.isEmpty() )
+    return QDomElement();
+
   QDomElement uoElem;
   switch ( node->op() )
   {
     case QgsExpression::uoMinus:
       uoElem = doc.createElement( "ogc:Literal" );
-      uoElem.appendChild( doc.createTextNode( "-" ) );
+      if ( node->operand()->nodeType() == QgsExpression::ntLiteral )
+      {
+        // operand expression already created a Literal node:
+        // take the literal value, prepend - and remove old literal node
+        uoElem.appendChild( doc.createTextNode( "-" + operandElem.text() ) );
+        doc.removeChild(operandElem);
+      }
+      else // not sure if this will ever happen
+      {
+        uoElem.appendChild( doc.createTextNode( "-" ) );
```

Comment:
@wonder-sk I kept that line to not break historical use of it...

I'm fine with an error message, as I think (espcially in the ogc/sld context) this will never be used/happen.

Should I change it and add it to this PR then?